cmd/server/internal/goreman/goreman.go

```
@@ -54,14 +55,46 @@ func readProcfile(content []byte) error {
 	return nil
 }
 
+// ProcDiedAction specifies the behaviour Goreman takes if a process exits
+// with a non-zero exit code.
+type ProcDiedAction int
+
+const (
+	// Shutdown will shutdown Goreman if any process shuts down with a
+	// non-zero exit code.
+	Shutdown ProcDiedAction = iota
+
+	// Ignore will continue running Goreman and will leave not restart the
+	// dead process.
+	Ignore
+)
+
+// procDiedAction is the ProcDiedAction to take. Goreman still is globals
+// everywhere \o/
+var procDiedAction ProcDiedAction
+
+type Options struct {
+	// RPCAddr is the address to listen for Goreman RPCs.
+	RPCAddr string
+
+	// ProcDiedAction specifies the behaviour to take when a process dies.
+	ProcDiedAction ProcDiedAction
+}
+
 // Start starts up the Procfile.
-func Start(rpcAddr string, contents []byte) error {
+func Start(contents []byte, opts Options) error {
 	err := readProcfile(contents)
 	if err != nil {
 		return err
 	}
-	if err := startServer(rpcAddr); err != nil {
-		return err
+	procDiedAction = opts.ProcDiedAction
```

Should we return an error here if the given `ProcDiedAction` is invalid rather than at runtime?

---

Yeah could validate it here, thought it would be messy but on reflection is better than my log.Fatal